meilisearch-error/src/lib.rs

```diff
@@ -85,26 +85,34 @@ impl Code {
 
         match self {
             // index related errors
-            CreateIndex => ErrCode::invalid("create_index", StatusCode::BAD_REQUEST),
+            // create index is thrown on internal error while creating an index.
+            CreateIndex => ErrCode::internal("create_index", StatusCode::BAD_REQUEST),
+            // thrown when an user want to creae an index twice with the same uid
             IndexAlreadyExists => ErrCode::invalid("index_already_exists", StatusCode::BAD_REQUEST),
+            // thrown when requesting an unexisting index
             IndexNotFound => ErrCode::invalid("index_not_found", StatusCode::NOT_FOUND), InvalidIndexUid => ErrCode::invalid("invalid_index_uid", StatusCode::BAD_REQUEST),
+            // internal error thrown when trying to open an index
             OpenIndex => ErrCode::internal("open_index", StatusCode::INTERNAL_SERVER_ERROR),
 
             // invalid state error
             InvalidState => ErrCode::internal("invalid_state", StatusCode::INTERNAL_SERVER_ERROR),
+            // thrown when no primary key has been set
             MissingPrimaryKey => ErrCode::internal("missing_primary_key", StatusCode::INTERNAL_SERVER_ERROR),
-            PrimaryKeyAlreadyPresent => ErrCode::internal("primary_key_already_present", StatusCode::INTERNAL_SERVER_ERROR),
+            // error thrown when trying to set an already existing primary key
+            PrimaryKeyAlreadyPresent => ErrCode::invalid("primary_key_already_present", StatusCode::BAD_REQUEST),
 
             // invalid document
             MaxFieldsLimitExceeded => ErrCode::invalid("max_field_limit_exceeded", StatusCode::BAD_REQUEST),
```

Comment:
```suggestion
            MaxFieldsLimitExceeded => ErrCode::invalid("max_fields_limit_exceeded", StatusCode::BAD_REQUEST),
```

Reply:
> For missing_primary_key, I assume that means MS cannot infer the primary key. Could we rename the code accordingly? Like could_not_infer_primary_key slightly_smiling_face So that we can clearly understand the difference between this error and the missing_document_id one

no objection on my end. @Kerollmops, @bidoubiwa, @eskombro ?

Reply:
I like the idea, I don't understand why we talk about a `document_id` instead of the normal term we use, primary key. I propose:

`could_not_infer_primary_key`
`missing_primary_key_in_document`